Source/DafnyLanguageServer.Test/GutterStatus/ConcurrentLinearVerificationGutterStatusTester.cs

```diff
@@ -44,14 +44,14 @@ public class ConcurrentLinearVerificationGutterStatusTester : LinearVerification
     // That way, it can rebuild the trace for every file independently.
     for (var i = 0; i < MaxSimultaneousVerificationTasks; i++) {
       result.Add(VerifyTrace(@"
- .  |  |  |  I  |  | :predicate F(i: int) {
- .  |  |  |  I  |  | :  false // Should not be highlighted in gutter.
- .  |  |  |  I  |  | :}
-    |  |  |  I  |  | :
- .  S [S][ ][I][S][ ]:method H()
- .  S [=][=][-][~][O]:  ensures F(1)
- .  S [=][=][-][~][=]:{//Replace: { assert false;
- .  S [S][ ][I][S][ ]:}", false, $"EnsuresManyDocumentsCanBeVerifiedAtOnce{i}.dfy", true, true, verificationStatusGutterReceivers[i]));
+ .  |  |  |  |  I  |  |  | :predicate F(i: int) {
+ .  |  |  |  |  I  |  |  | :  false // Should not be highlighted in gutter.
+ .  |  |  |  |  I  |  |  | :}
+    |  |  |  |  I  |  |  | :
+ .  .  S [S][ ][I][I][S][ ]:method H()
+ .  .  S [=][=][-][-][~][O]:  ensures F(1)
+ .  .  S [=][=][-][-][~][=]:{//Replace: { assert false;
+ .  .  S [S][ ][I][I][S][ ]:}", false, $"EnsuresManyDocumentsCanBeVerifiedAtOnce{i}.dfy", true, true, verificationStatusGutterReceivers[i]));
```

Comment:
When the server detects that it's testing gutter icons, it will translate each module one by one, and then schedule tasks to be verified one by one.

Outside of this testing mode, the server will translate and schedule tasks one by one, so a task can begin verification while another is still translating. This mode of operation, while correct, makes it impossible to use the current testing format since that format assumes too much determinism. I didn't want to change the tests too much so I introduced a special mode on the server that allows the current tests to pass.

Reply:
Ok thanks for the explanation, it makes sense. Glad you introduced a more deterministic testing mode.